ui_testing/testcases/basic_tests/test06_orders.py

```diff
@@ -189,28 +188,28 @@ def test004_archive_order(self):
             rows = self.order_page.result_table()
             self.assertEqual(len(rows), 1)
 
-    # will continue with us
     def test005_restore_archived_orders(self):
         """
         Restore Order
         I can restore any order successfully
         LIMS-4374
         """
-        self.base_selenium.LOGGER.info(' + Get Archived orders ')
         self.order_page.get_archived_items()
-        self.base_selenium.LOGGER.info(' + Select Rows ')
-        selected_orders_data, _ = self.order_page.select_random_multiple_table_rows()
-
-        self.base_selenium.LOGGER.info(' + Restore Selected Rows ')
-        self.order_page.restore_selected_items()
-        self.base_selenium.LOGGER.info(' + Get Active orders')
-        self.order_page.get_active_items()
-        for selected_order_data in selected_orders_data:
-            self.base_selenium.LOGGER.info(' + Order with analysis number =  {} restored successfully?'.format(
-                selected_order_data['Analysis No.']))
-            self.assertTrue(self.order_page.is_order_exist(
-                value=selected_order_data['Analysis No.']))
-
+        orders, payload = self.orders_api.get_all_orders(deleted=1)
+        order_data = random.choice(orders['orders'])
+
+        order_no = order_data['orderNo']
+        self.order_page.apply_filter_scenario(filter_element='orders:filter_order_no', filter_text=order_no, field_type='text')
+        suborders_data = self.order_page.get_child_table_data(index=0)
+        self.order_page.restore_table_suborder(index=0)
+        self.base_selenium.LOGGER.info('make sure that suborder is restored successfully')
+        if len(suborders_data) > 1:
+            suborder_data_after_restore = self.order_page.get_table_data()
+            self.assertNotEqual(suborder_data_after_restore[0], suborders_data[0])
+        else:
+            table_records_count = len(self.order_page.result_table()) -1
+            self.assertEqual(table_records_count, 0)
+      
```

Comment:
I think you should assert that suborder restored by checking that it is in active table

Reply:
if order has another suborder, this assertion will be true even if suborder not restored
so, you need to make assertion by analysis no not order no

Reply:
Done

Reply:
@deenaahmed `self.assertTrue(ANY_VALID_VALUE)` is true so u need to check that `result` data is matching with the suborder data